src/features/formData/FormDataWrite.tsx

```diff
@@ -210,7 +215,13 @@ function FormDataEffects() {
   // Marking the document as having unsaved changes. The data attribute is used in tests, while the beforeunload
   // event is used to warn the user when they try to navigate away from the page with unsaved changes.
   useEffect(() => {
-    document.body.setAttribute('data-unsaved-changes', hasUnsavedChanges.toString());
+    if (hasUnsavedChanges) {
+      document.body.setAttribute('data-unsaved-changes', 'true');
+    } else {
+      setTimeout(() => {
+        document.body.setAttribute('data-unsaved-changes', 'false');
+      }, 10);
+    }
```

Comment:
Could this `setTimeout` when setting unsaved changes to false be problematic?

What if it changes back to true before the timeout fires? Would it not then be set to false in the document body once the timer expires, even though its suppose to be true?

Maybe the timeout should be kept track of, so it can be cleared when setting to true?

Reply:
Absolutely! :raised_hands: Changed that now (but haven't pushed yet, as tests are currently in disarray while I fix some stuff properly).